app/components/create-player-form.js

```diff
@@ -0,0 +1,35 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class CreatePlayerFormComponent extends Component {
+  @service store;
+
+  @tracked
+  nickname = '';
+
+  @action
+  async createPlayerHandler(event) {
+    // Stops submit event from reloading page
+    event.preventDefault();
+
+    const { nickname, args } = this;
+
+    try {
+      if (!nickname.length) return;
+
+      args.switchModeHandler();
+
+      this.nickname = '';
+
+      const createdPlayer = this.store.createRecord('player', {
+        nickname: nickname,
+      });
+
+      await createdPlayer.save();
```

Comment:
We try to avoid await in actions. I know that in this context await is not bad but we prefer to use task from https://github.com/machty/ember-concurrency (There is a lot of benefit by using it like isLoading status out of the box)

Please try to refactor using a task from ember-concurrency. If you encounter some problems let me know